boards/arm/lpcxpresso55s69/lpcxpresso55s69-pinctrl.dtsi

```diff
@@ -0,0 +1,94 @@
+/*
+ * NOTE: File generated by lpc_cfg_utils.py
+ * from LPCXpresso55S69.mex
+ *
+ * Copyright (c) 2022, NXP
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <nxp/lpc/LPC55S69JBD100-pinctrl.h>
+
+&pinctrl {
+	flexcomm0_usart: flexcomm0_usart {
+		group0 {
+			pinmux = <FC0_TXD_SCL_MISO_WS_PIO0_30>,
+				<FC0_RXD_SDA_MOSI_DATA_PIO0_29>;
+			slew-rate = "standard";
+		};
+	};
+
+	flexcomm2_usart: flexcomm2_usart {
+		group0 {
+			pinmux = <FC2_RXD_SDA_MOSI_DATA_PIO1_24>,
+				<FC2_TXD_SCL_MISO_WS_PIO0_27>;
+			slew-rate = "standard";
+		};
+	};
+
+	flexcomm4_i2c: flexcomm4_i2c {
+		group0 {
+			pinmux = <FC4_TXD_SCL_MISO_WS_PIO1_20>,
+				<FC4_RXD_SDA_MOSI_DATA_PIO1_21>;
+			slew-rate = "standard";
+		};
+	};
+
+	flexcomm4_i2c_fxos8700: flexcomm4_i2c_fxos8700 {
+		group0 {
+			pinmux = <GPIO19_PIO1_19>;
+			bias-pull-up;
+			slew-rate = "standard";
+		};
+		group1 {
+			pinmux = <FC4_TXD_SCL_MISO_WS_PIO1_20>,
+				<FC4_RXD_SDA_MOSI_DATA_PIO1_21>;
+			slew-rate = "standard";
+		};
+	};
+
+	flexcomm6_i2s: flexcomm6_i2s {
+		group0 {
+			pinmux = <FC6_RXD_SDA_MOSI_DATA_PIO1_13>;
+			bias-pull-up;
+			slew-rate = "standard";
+		};
+	};
+
+	flexcomm7_i2s: flexcomm7_i2s {
+		group0 {
+			pinmux = <FC7_TXD_SCL_MISO_WS_PIO0_19>,
+				<FC7_RXD_SDA_MOSI_DATA_PIO0_20>;
+			bias-pull-up;
+			slew-rate = "standard";
+		};
+	};
+
+	gpio0_default: gpio0_default {
+		group0 {
+			pinmux = <GPIO5_PIO0_5>;
+			bias-pull-up;
+			slew-rate = "standard";
+		};
+	};
+
+	gpio1_default: gpio1_default {
+		group0 {
+			pinmux = <GPIO9_PIO1_9>,
+				<GPIO18_PIO1_18>;
+			bias-pull-up;
+			slew-rate = "standard";
+		};
+	};
```

Comment:
note: I think this belongs to the GPIO domain, not pinctrl. Can you clarify this?

Reply:
That's a good point. I was under the impression that for GPIO devices, pinctrl drivers would handle the GPIO pin selection, and GPIO drivers themselves would handle setting pin state, such as input vs output.

In my mind the division was on a register level: the pinctrl driver would be responsible for writes to the pinmuxing `IOCON` peripheral, and the gpio driver would be responsible for writes to the `GPIO` peripheral. So there would still need to be a pinctrl group for the GPIO pins, to select them in the `IOCON` pinmux

Reply:
After looking at the GPIO driver for Kinetis I see what you mean. I've updated the LPC GPIO driver for pin control in this PR

Reply:
note: STM32 has opened a PR to support static GPIO configuration via pinctrl, it may be useful for some edge cases, even though GPIO API is the way to go. I'll document that as part of the pinctrl guide. ref. https://github.com/zephyrproject-rtos/zephyr/pull/43961